mcs_benchmark_data/pipelines/commonsense_qa/commonsense_qa_submission_transformer.py

```diff
@@ -14,72 +15,68 @@
 
 class CommonsenseQaSubmissionTransformer(_Transformer):
 
+    __URI_BASE = "benchmark:commonsense_qa"
+    __BENCHMARK_SCORE_CLASSES={
+        "TestScore": TestScore,
+        "DevScore": DevScore
+    }
+
     def transform(
         self,
         *,
         system: str,
-        benchmark_json_file_path: Path,
-        dev_jsonl_file_path: Path,
-        test_jsonl_file_path: Path,
-        train_jsonl_file_path: Path,
         submission_data_jsonl_file_path: Path,
-        submission_jsonl_file_paths: Tuple[Path, ...]
+        submission_jsonl_file_paths: Tuple[Path, ...],
+        **kwds
     ) -> Generator[_Model, None, None]:
 
         # Yield submissions
         # Assumes file name in form "*_[systemname]_submission.jsonl" (e.g. dev_rand_split_roberta_submission.jsonl)
-        submission_data_jsonl = open(submission_data_jsonl_file_path)

```

Comment:
There won't be a single CommonsenseQaSubmissionTransformer (or -Extractor), because every submission transformer+extractor pair is system-specific. So CommonsenseQaRobertaSubmissionTransformer. That's a mouthful.

Reply:
This is true even though most of the code will be copied (similar) for both?

Reply:
There's nothing saying you can't reuse code, merely that each submission+benchmark combination will have its own pipeline subclass. If it makes sense to reuse an extractor or transformer parts of them (with composition or inheritance, preferring the former) then do that.